### DISCUSSION TITLE
How to create many things in one hook only? Why this code doesn't work?

### DISCUSSION THREAD

**Discussion by frederikhors** [uptrace/bun#380] [Q&A] [answered]
I would like to know how to create many thing for a table in one hook.

Example code that doesn't work:

```go
var _ bun.AfterCreateTableHook = (*Player)(nil)

func (*Player) AfterCreateTable(_ context.Context, query *bun.CreateTableQuery) error {
	query.ForeignKey(`("team_id") REFERENCES "teams" ("id")`)

	_, err := query.DB().Exec(`CREATE TRIGGER someTrigger BEFORE UPDATE ON "players" FOR EACH ROW EXECUTE PROCEDURE someTrigger()`)
	if err != nil {
		return err
	}

	return nil
}
```

This doesn't create the ForeignKey, why?

If I remove the `Exec()` part it creates the foreign key.

**Comment by vmihailenco** [comment, marked as answer]
`AfterCreateTable` is run after the table is created so `query.ForeignKey(`("team_id") REFERENCES "teams" ("id")`)` does not make any difference / is ignored. You should move it to `BeforeCreateTable`.
